src/main/java/org/oscarehr/inboxhub/display/ManageInboxhubAction.java

```diff
@@ -0,0 +1,116 @@
+
+    public ActionForward undefined(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) {
+        return mapping.findForward("error");
+    }
+
+    public ActionForward displayInboxForm(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) {
+        LoggedInInfo loggedInInfo = LoggedInInfo.getLoggedInInfoFromSession(request);
+        if (!securityInfoManager.hasPrivilege(loggedInInfo, "_lab", SecurityInfoManager.READ, null)) {
+            return mapping.findForward("unauthorized");
+        }
+
+        LabDataController labDataController = new LabDataController();
+        String unclaimed = (String) request.getParameter("unclaimed");
+        InboxhubQuery query = (InboxhubQuery) form;
+        labDataController.setInboxFormQueryUnclaimed(query, unclaimed);
+        request.setAttribute("query", query);
+
+        labDataController.sanitizeInboxFormQuery(loggedInInfo, query, null, null);
+        CategoryData categoryData = labDataController.getCategoryData(query);
+        int totalResultsCount = labDataController.getTotalResultsCountBasedOnQuery(query, categoryData);
+
+        request.setAttribute("totalResultsCount", totalResultsCount);
+        request.setAttribute("viewMode", query.getViewMode());
+        request.setAttribute("categoryData", categoryData);
+        return mapping.findForward("success");
+    }
+
+    public ActionForward displayInboxList(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) {
+        LoggedInInfo loggedInInfo = LoggedInInfo.getLoggedInInfoFromSession(request);
+        if (!securityInfoManager.hasPrivilege(loggedInInfo, "_lab", SecurityInfoManager.READ, null)) {
+            return mapping.findForward("unauthorized");
+        }
+
+        fetchLabData(form, request);
+        return mapping.findForward("displayList");
+    }
+
+    public ActionForward displayInboxView(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) {
+        LoggedInInfo loggedInInfo = LoggedInInfo.getLoggedInInfoFromSession(request);
+        if (!securityInfoManager.hasPrivilege(loggedInInfo, "_lab", SecurityInfoManager.READ, null)) {
+            return mapping.findForward("unauthorized");
+        }
+
+        fetchLabData(form, request);
+        return mapping.findForward("displayView");
+    }
+
+    private void fetchLabData(ActionForm form, HttpServletRequest request) {
+        LoggedInInfo loggedInInfo = LoggedInInfo.getLoggedInInfoFromSession(request);
+
+        String page = request.getParameter("page");
+        String pageSize = request.getParameter("pageSize");
+        String demographicFilter = request.getParameter("demographicFilter");
+        String typeFilter = request.getParameter("typeFilter");
+
+        InboxhubQuery query = (InboxhubQuery) form;
+        query.setPage(Integer.parseInt(page));
```

Comment:
No need to add additional exception handling here. If the code breaks due to malformed input, the root cause should be addressed directly to prevent this from happening rather than relying on try-catch blocks for validation.